gateway/gateway.go

```diff
@@ -282,54 +323,85 @@ func (g *gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 	// errors will occur.
 	// Unexpected behavior will also occur if stopGateway() is called when the gateway hasn't been
 	// started, so only call stopGateway if this module already started the gateway.
-	if g.started {
-		g.reset(ctx)
+	if err := g.resetConcentrators(ctx); err != nil {
+		return err
 	}
 
 	// workers and process are restarted during reconfigure
-	g.closeProcess()
-	if g.workers != nil {
-		g.workers.Stop()
+	g.closeProcesses()
+
+	// maintain devices and lastReadings through reconfigure.
+	if g.devices == nil {
+		g.devices = make(map[string]*node.Node)
 	}
 
-	cfg, err := getNativeConfig(conf)
-	if err != nil {
-		return err
+	if g.lastReadings == nil {
+		g.lastReadings = make(map[string]interface{})
 	}
 
+	g.concentrators = make([]*concentrator, 0)
+
+	switch conf.Model {
+	case ModelRak7391:
+		cfg, err := getMultiConcentratorConfig(conf)
+		if err != nil {
+			return err
+		}
+		if err := g.reconfigureMultiConcentrator(ctx, deps, cfg); err != nil {
+			return err
+		}
+	default:
+		cfg, err := getNativeConfig(conf)
+		if err != nil {
+			return err
+		}
+		if err := g.reconfigureSingleConcentrator(ctx, deps, cfg); err != nil {
+			return err
+		}
+	}
+
+	g.workers = utils.NewBackgroundStoppableWorkers(g.receivePackets)
+
+	return nil
+}
+
+func (g *gateway) reconfigureSingleConcentrator(ctx context.Context, deps resource.Dependencies, cfg *Config) error {
 	board, err := board.FromDependencies(deps, cfg.BoardName)
 	if err != nil {
 		return err
 	}
 
-	// maintain devices and lastReadings through reconfigure.
-	if g.devices == nil {
-		g.devices = make(map[string]*node.Node)
+	region := regions.GetRegion(cfg.Region)
+
+	switch region {
+	case regions.US, regions.Unspecified:
+		g.logger.Infof("configuring gateway for US915 band")
+		g.regionInfo = regions.RegionInfoUS
+		g.region = regions.US
+	case regions.EU:
+		g.logger.Infof("configuring gateway for EU868 band")
+		g.regionInfo = regions.RegionInfoEU
+		g.region = regions.EU
 	}
```

Comment:
could this check live in the main Reconfigure function? gut feeling is no because the different configs